globus_sdk/base.py

```diff
@@ -7,6 +7,46 @@
 from globus_sdk import config
 
 
+class GlobusError(Exception):
+    def __init__(self, r, *args, **kw):
+        self._underlying_response = r
+        self.http_status = r.status_code
+        if "application/json" in r.headers["Content-Type"]:
```

Comment:
Question: should we even be checking `Content-Type` here?
A JSON response could be returned without this header, or with a wrong header (apparently).

Should we just *always* try to load from JSON, catch failed parsing, and fallback to text?